addons/Dexie.Observable/src/Dexie.Observable.js

```diff
@@ -256,18 +256,23 @@ function Observable(db) {
                             currentMaster.isMaster = 0;
                             return db._syncNodes.put(currentMaster);
                         }
-                    }).then(()=>{
-                        // Add our node to DB and start subscribing to events
-                        return db._syncNodes.add(mySyncNode.node).then(function() {
-                            Observable.on('latestRevisionIncremented', onLatestRevisionIncremented); // Wakeup when a new revision is available.
-                            Observable.on('beforeunload', onBeforeUnload);
-                            Observable.on('suicideNurseCall', onSuicide);
-                            Observable.on('intercomm', onIntercomm);
-                            // Start polling for changes and do cleanups:
-                            pollHandle = setTimeout(poll, LOCAL_POLL);
-                            // Start heartbeat
-                            heartbeatHandle = setTimeout(heartbeat, HEARTBEAT_INTERVAL);
-                        });
+                    });
+                Promise.all([currentMaster]);
```

Comment:
The intent behind this serialization was to attempt to keep the parent transaction 'active', as I have a sense that it is being closed as soon as the read operation (`where`) completes, and that as a result the `db._syncNodes` write operations fail later on.

I'm not super familiar with JavaScript promises so I'm not completely confident that this is still performing the correct set of operations in the correct order..

Reply:
(this approach was indeed incorrect, as discovered by the `dexie-observable` unit test coverage)